rotkehlchen/chain/ethereum/modules/thegraph/balances.py

```diff
@@ -0,0 +1,122 @@
+            evm_inquirer: 'EvmNodeInquirer',
+            chain_id: 'CHAIN_IDS_WITH_BALANCE_PROTOCOLS',
+    ):
+        super().__init__(
+            database=database,
+            evm_inquirer=evm_inquirer,
+            chain_id=chain_id,
+            counterparty=CPT_THEGRAPH,
+        )
+        self.grt = A_GRT.resolve_to_evm_token()
+        self.staking_contract = string_to_evm_address('0xF55041E37E12cD407ad00CE2910B8269B01263b9')
+
+    def get_deposit_events(self) -> set[tuple[HistoryEventType, HistoryEventSubType]]:
+        return {(HistoryEventType.STAKING, HistoryEventSubType.DEPOSIT_ASSET)}
+
+    def query_balances(self) -> 'BalancesType':
+        balances: BalancesType = defaultdict(lambda: defaultdict(Balance))
+        grt_price = Inquirer().find_usd_price(A_GRT)
+
+        # fetch deposit events
+        addresses_with_deposits = self.addresses_with_deposits(
+            product=EvmProduct.STAKING,
+            deposit_events=self.get_deposit_events(),
+        )
+        # remap all events into a list that will contain all pairs (delegator, indexer)
+        delegations = set()
+        for delegator, event_list in addresses_with_deposits.items():
+            for event in event_list:
+                if event.extra_data is None:
+                    continue
+                if (indexer := event.extra_data.get('indexer')) is not None:
+                    delegations.add((delegator, indexer))
+        delegations_unique = list(delegations)
+
+        staking_contract = self.evm_inquirer.contracts.contract(self.staking_contract)
+        chunk_size, call_order = get_chunk_size_call_order(self.evm_inquirer)
+
+        # query how many shares delegator currently have at the indexer/pool
+        all_shares = self.evm_inquirer.multicall(
+            calls=[(
+                staking_contract.address,
+                staking_contract.encode(
+                    method_name='getDelegation',
+                    arguments=[indexer, delegator],
+                ),
+            ) for delegator, indexer in delegations_unique],
+            call_order=call_order,
+            calls_chunk_size=chunk_size,
+        )
+
+        # query pool data to get current total amount of shares and tokens in that pool
+        pools = self.evm_inquirer.multicall(
+            calls=[(
+                staking_contract.address,
+                staking_contract.encode(
+                    method_name='delegationPools',
+                    arguments=[indexer],
+                ),
+            ) for _, indexer in delegations_unique],
+            call_order=call_order,
+            calls_chunk_size=chunk_size,
+        )
+
+        for idx, delegation in enumerate(delegations_unique):
+            delegator = delegation[0]
+            shares_response = staking_contract.decode(
+                all_shares[idx],
+                'getDelegation',
+                arguments=[delegator, delegation[1]],
+            )[0]
```

Comment:
If I'm correct you don't use shares_response anymore. Instead of doing `shares = shares_response[0]` just do

```suggestion
            shares = staking_contract.decode(
                all_shares[idx],
                'getDelegation',
                arguments=[delegator, delegation[1]],
            )[0][0]
```

or is cleaner if you do

```suggestion
            shares, _ = staking_contract.decode(
                all_shares[idx],
                'getDelegation',
                arguments=[delegator, delegation[1]],
            )[0]
```

you wil need as many `_` as values are returned by the function